src/GraphQL.Conventions/Adapters/Resolvers/EventStreamResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿using GraphQL.Conventions.Types.Descriptors;
+using GraphQL.Resolvers;
+using GraphQL.Subscription;
+using GraphQL.Types;
+using System;
+using System.Threading.Tasks;
+
+namespace GraphQL.Conventions.Adapters.Resolvers
+{
+    internal class EventStreamResolver : FieldResolver, IEventStreamResolver
+    {
+        public EventStreamResolver(GraphFieldInfo fieldInfo) : base(fieldInfo)
+        {
+        }
+
+        public IObservable<object> Subscribe(ResolveEventStreamContext context)
+        {
+            var result = Resolve(context);
+            if (result is Task<object>)
+            {
+                result = (result as Task<object>).Result;

```

Comment:
Have you done any testing with more complex `Task` sources here?

Reply:
I guess no I a dont know exactly what you mean.

Reply:
In other words, if you have something that returns `Task<IObservable<object>>`, does the `.Result` behave as expected here?

Reply:
I guess it should. As long as we return an `IObservable<object>` the subscription mechanics should work. And `Task<IObservable>.Result` returns an `IObservable<object>`. Maybe I dont get it. I dont know what the test should test.